scripts/safe_template_linter.py

```diff
@@ -56,6 +56,79 @@ def _load_file(self, file_full_path):
         return file_contents.decode(encoding='utf-8')
 
 
+def _find_closing_char_index(start_delim, open_char, close_char, template, start_index, num_open_chars=0, strings=None):
+    """
+    Finds the index of the closing char that matches the opening char.
+
+    For example, this could be used to find the end of a Mako expression,
+    where the open and close characters would be '{' and '}'.
+
+    Arguments:
+        start_delim: If provided (e.g. '${' for Mako expressions), the
+            closing character must be found before the next start_delim.
+        open_char: The opening character to be matched (e.g '{')
+        close_char: The closing character to be matched (e.g '}')
+        template: The template to be searched.
+        start_index: The start index of the last open char.
+        num_open_chars: The current number of open chars.
+        strings: A list of ParseStrings already parsed
+
+    Returns:
+        A dict containing the following:
+            close_char_index: The index of the closing character, or -1 if
+            unparseable.
+            strings: a list of ParseStrings
+
+    """
+    strings = [] if strings is None else strings
+    unparseable_result = {'close_char_index': -1, 'strings': []}
+    close_char_index = template.find(close_char, start_index)
```

Comment:
How would this handle a closing character inside a string before the real closing character? Would it match it incorrectly? e.g. when matching parens, what would happen with:

    this.$el.text('Exit Music (For a Film)');